bundles/org.openhab.binding.denonmarantz/src/main/java/org/openhab/binding/denonmarantz/internal/connector/http/DenonMarantzHttpConnector.java

```diff
@@ -295,20 +378,39 @@ private boolean setConfigProperties() throws IOException {
         return (deviceinfo != null);
     }
 
-    private void refreshHttpProperties() throws IOException {
+    private void refreshHttpProperties() throws TimeoutException, ExecutionException, InterruptedException {
         logger.trace("Refreshing Denon status");
 
-        updateMain();
-        updateMainZone();
+        if (legacyApiSupported) {
+            updateMain();
+            updateMainZone();
+        }
+
+        if (!legacyApiSupported) {
+            updateMainZoneByAppCommand();
+        }
+
         updateSecondaryZones();
         updateDisplayInfo();
     }
 
     @Nullable
-    private <T> T getDocument(String uri, Class<T> response) throws IOException {
+    private <T> T getDocument(String uri, Class<T> response)
+            throws TimeoutException, ExecutionException, InterruptedException, HttpCommunicationException {
         try {
-            String result = HttpUtil.executeUrl("GET", uri, REQUEST_TIMEOUT_MS);
-            logger.trace("result of getDocument for uri '{}':\r\n{}", uri, result);
+            Request request = httpClient.newRequest(uri).timeout(REQUEST_TIMEOUT_MS, TimeUnit.MILLISECONDS)
+                    .method(HttpMethod.GET);
+
+            ContentResponse contentResponse = request.send();
+
+            String result = contentResponse.getContentAsString();
+            int status = contentResponse.getStatus();
+
+            logger.trace("result of getDocument for uri '{}' (status code {}):\r\n{}", uri, status, result);
+
+            if (!HttpStatus.isSuccess(status)) {
+                throw new HttpCommunicationException("Error retrieving document for uri '{}'", status);
```

Comment:
In the first parameter, I believe you have to provide the uri and not `{}`.

Reply:
Fixed.